salt/metalk8s/addons/dex/config/dex.yaml.j2

```diff
@@ -54,7 +65,7 @@ spec:
     - id: metalk8s-ui
       name: MetalK8s UI
       redirectURIs:
-      - https://{{ grains.metalk8s.control_plane_ip }}:8443/
+      - https://{{ grains.metalk8s.control_plane_ip }}:8443{{ metalk8s_ui_config.spec.basePath }}
```

Comment:
ditto:

```jinja
...:8443/{{ basePath.strip('/') }}/
```

or, if we don't care about the trailing slash (but will keep it if specified):

```jinja
...:8443/{{ basePath.lstrip('/') }}
```

Reply:
done, I introduced a `normalised_base_path` in the templates which ensures the basePath to be of the shape `/{optional_path}`